src/porepy/models/verification_setups/terzaghi.py

```diff
@@ -0,0 +1,819 @@
+                    * np.exp((-((2 * i - 1) ** 2)) * (np.pi**2 / 4) * nondim_t)
+                )
+            p = (4 / np.pi) * F * sum_series
+
+        return p
+
+    def consolidation_degree(self, t: number) -> float:
+        """Compute exact degree of consolidation.
+
+        Parameters:
+            t : Time [s].
+
+        Returns:
+            Degree of consolidation for the given time.
+
+        """
+        t_nondim = self.setup.nondim_time(t)
+        n = self.setup.params.get("upper_limit_summation", 1000)
+
+        if t == 0:  # initially, the soil is unconsolidated
+            deg_cons = 0.0
+        else:
+            sum_series = 0
+            for i in range(1, n + 1):
+                sum_series += (
+                    1
+                    / ((2 * i - 1) ** 2)
+                    * np.exp(-((2 * i - 1) ** 2) * (np.pi**2 / 4) * t_nondim)
+                )
+            deg_cons = 1 - (8 / (np.pi**2)) * sum_series
+
+        return deg_cons
+
+
+class ModifiedDataSavingMixin(pp.DataSavingMixin):
+    """Mixin class to save relevant data."""
+
+    pressure_variable: str
+    """Key to access the pressure variable."""
+
+    displacement_variable: str
+    """Key to access the displacement variable"""
+
+    nondim_time: Callable[[number], number]
+    """Method that non-dimensionalizes time. The method is provided by the mixin class
+    :class:`SetupUtilities`.
+
+    """
+
+    nondim_length: Callable[[np.ndarray], np.ndarray]
+    """Method that non-dimensionalises length. The method is provided by the mixin
+    class :class:`SetupUtilities`.
+
+    """
+
+    nondim_pressure: Callable[[np.ndarray], np.ndarray]
+    """Method that non-dimensionalises pressure. The method is provided by the mixin
+    class :class:`SetupUtilities`.
+
+    """
+
+    exact_sol: ExactSolution
+    """Exact solution object."""
+
+    results: list[SaveData]
+    """List of :class:`SaveData` objects containing the results of the verification."""
+
+    numerical_consolidation_degree: Callable[[np.ndarray, np.ndarray], number]
+    """Method that computes the numerical degree of consolidation. The method is
+    provided by the mixin class :class:`SetupUtilities`.
+
+    """
+
+    relative_l2_error: Callable[[grid, np.ndarray, np.ndarray, bool, bool], number]
+    """Method that computes the discrete relative L2-error. The method is provided by
+    the mixin class:class:`porepy.models.verification_setups.VerificationUtils`.
+
+    """
+
+    def save_data(self) -> None:
+        """Save data to the `results` list.
+
+        Note:
+            Data will be appended to the ``results`` list only if the current time
+            matches a time from ``self.params["stored_times"]``. By default,
+            only the data for the final simulation time is stored.
+
+        """
+        t = self.time_manager.time
+        tf = self.time_manager.time_final
+        if any(np.isclose(t, self.params.get("stored_times", [tf]))):
```

Comment:
Could we use time_manager's schedule to infer "stored_times"?

Reply:
Yes, it should be doable with the schedule.